gnuradio-runtime/lib/top_block_impl.cc

```diff
@@ -129,6 +132,21 @@ namespace gr {
   void
   top_block_impl::wait()
   {
+    do {
```

Comment:
It's unneeded, and optimized away by the compiler anyway.  The local scope for the scoped_lock is provided by the opening and closing braces on lines 137 and 143.